js/src/forum/components/CommentPost.js

```diff
@@ -41,48 +40,44 @@ export default class CommentPost extends Post {
 
     this.subtree.check(
       () => this.cardVisible,
-      () => this.isEditing()
+      () => this.isEditing(),
+      () => this.revealContent
     );
   }
 
   content() {
-    // Note: we avoid using JSX for the <ul> below because it results in some
-    // weirdness in Mithril.js 0.1.x (see flarum/core#975). This workaround can
-    // be reverted when we upgrade to Mithril 1.0.
-    return super
-      .content()
-      .concat([
-        <header className="Post-header">{m('ul', listItems(this.headerItems().toArray()))}</header>,
-        <div className="Post-body">
-          {this.isEditing() ? <div className="Post-preview" config={this.configPreview.bind(this)} /> : m.trust(this.props.post.contentHtml())}
-        </div>,
-      ]);
+    return super.content().concat([
+      <header className="Post-header">
+        <ul>{listItems(this.headerItems().toArray())}</ul>
+      </header>,
+      <div className="Post-body">{this.isEditing() ? <CommentPostPreview /> : m.trust(this.attrs.post.contentHtml())}</div>,
+    ]);
   }
 
-  config(isInitialized, context) {
-    super.config(...arguments);
+  onupdate(vnode) {
+    super.onupdate();
 
-    const contentHtml = this.isEditing() ? '' : this.props.post.contentHtml();
+    const contentHtml = this.isEditing() ? '' : this.attrs.post.contentHtml();
 
     // If the post content has changed since the last render, we'll run through
     // all of the <script> tags in the content and evaluate them. This is
     // necessary because TextFormatter outputs them for e.g. syntax highlighting.
-    if (context.contentHtml !== contentHtml) {
+    if (vnode.dom.contentHtml !== contentHtml) {
       this.$('.Post-body script').each(function () {
         eval.call(window, $(this).text());
       });
     }
 
-    context.contentHtml = contentHtml;
+    vnode.dom.contentHtml = contentHtml;
```

Comment:
Hm yeah not sure why that's being stored there now. Maybe because of `context`? Though to be honest I never understood what `context` is. It might have been the VDOM.

Reply:
Yeah, this was moved over because I didn't understand what context did (and why).